### PR TITLE
Add endpoint to get user names for a given directory element

### PR DESCRIPTION
**Pull request by jonenst**
TODO tests

### CODE REVIEW

File: src/main/java/org/gridsuite/explore/server/services/UserIdentityService.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    public String getUserIdentities(List<String> subs) {
+        String path = UriComponentsBuilder.fromPath(DELIMITER + USER_IDENTITY_API_VERSION + USERS_IDENTITY_PATH)
+                .buildAndExpand(String.join(",", subs)).toUriString();
+        try {
+            return restTemplate.getForObject(userIdentityServerBaseUri + path, String.class);
+        } catch (HttpStatusCodeException e) {
+            if (e.getStatusCode().value() == 404) {
+                return null; // no profile == unlimited import
```
[the remark this answers is not in the document]
Comment (sBouzols) [reply src/main/java/org/gridsuite/explore/server/services/UserIdentityService.java]: was a comment from UserAdminService, my bad, removed

File: src/test/java/org/gridsuite/explore/server/UserIdentityTest.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+                    return new MockResponse()
+                            .setResponseCode(200)
+                            .setHeader("Content-Type", "application/json")
+                            .setBody(objectMapper.writeValueAsString(elementAttributesList));
+                }
+                return new MockResponse().setResponseCode(404);
+            }
+        });
+    }
```
[the remark this answers is not in the document]
Comment (sBouzols) [reply src/test/java/org/gridsuite/explore/server/UserIdentityTest.java]: done

File: src/test/java/org/gridsuite/explore/server/UserIdentityTest.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+    @Test
+    void testGetSubIdentity() throws Exception {
+        mockMvc.perform(get(BASE_URL)
+                        .param("ids", ELEMENT_UUID.toString()))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON));
```
[the remark this answers is not in the document]
Comment (sBouzols) [reply src/test/java/org/gridsuite/explore/server/UserIdentityTest.java]: done

File: src/main/java/org/gridsuite/explore/server/services/UserIdentityService.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        this.restTemplate = restTemplate;
+    }
+
+    public String getUsersIdentities(List<String> subs) {
+        String path = UriComponentsBuilder.fromPath(DELIMITER + USER_IDENTITY_API_VERSION + USERS_IDENTITY_PATH)
+                .buildAndExpand(String.join(",", subs)).toUriString();
+        try {
+            return restTemplate.getForObject(userIdentityServerBaseUri + path, String.class);
+        } catch (HttpStatusCodeException e) {
+            if (e.getStatusCode().value() == 404) {
```
[the remark this answers is not in the document]
Comment (sBouzols) [reply src/main/java/org/gridsuite/explore/server/services/UserIdentityService.java]: I removed this control

**Review by sBouzols — APPROVED** [review]

**Review by klesaulnier — APPROVED** [review]

**Pull request merged**